pkg/provider/diff.go

```diff
@@ -78,6 +78,31 @@ var forceNew = _groups{
 var core = _versions{
 	"v1": _kinds{
 		"ConfigMap": properties{".binaryData", ".data"},
+		"PersistentVolume": append(
```

Comment:
Let's link the [volume types doc](https://kubernetes.io/docs/concepts/storage/volumes/#types-of-volumes).

---

Any chance we can programmatically pull this list from the API?

 e.g.

`kubectl explain pod.spec.volumes` shows all volume types so it seems like API's should be possible. Thoughts?

Reply:
I pulled this info from the API reference, but am not sure of a reliable way to do so programmatically. I included everything that had a Kind name ending in `VolumeSource`, but these Kinds don't seem to share a common interface.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#volume-v1-core

Reply:
Not a fan of reflect in Go due to its perf cost, but is it worth considering asserting the various interfaces to approach this programmatically. Is this viable?

Reply:
Since the rest of this file is also manually curated for now, I'm inclined to defer further work here. We need to figure out a more robust approach at some point, but that's probably going to involve upstream work to fix the k8s OpenAPI spec.

Reply:
#342